### ISSUE TITLE
Is Augmentation auto enabled?

### ISSUE THREAD

**Issue by darouwan** [ultralytics/ultralytics#23110]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I found some augmentation config in https://docs.ultralytics.com/guides/yolo-data-augmentation/#using-a-configuration-file has default value, and it looks auto enabled in training. However, I also found there is a `augment` parameter in train method, and I cannot find document related to it. So the data augmentation is auto enabled or not?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @darouwan, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get unblocked quickly, and an Ultralytics engineer will also assist soon 🙌

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by onuralpszr** [comment]
@darouwan  I pressume you meant for "augmentations"  and that's ONLY usable in python API not in CLI 

You can see in here : https://docs.ultralytics.com/modes/train/#augmentation-settings-and-hyperparameters

"Custom Albumentations transforms for advanced data augmentation (Python API only). Accepts a list of transform objects for specialized augmentation needs."

And for usage 

https://docs.ultralytics.com/guides/yolo-data-augmentation/#python-api

I hope that helps.

**Comment by darouwan** [comment]
@onuralpszr  Yes, I mean the augmentation function in python API. 
So if I want to enable data augmentation with default augmentation setting, do I need to set `augment=True` in train method explictly? 

https://github.com/ultralytics/ultralytics/blob/932c069b0273e97eea46745e3bc5bb73dcd9cecd/ultralytics/cfg/default.yaml#L65

**Comment by onuralpszr** [comment]
> [@onuralpszr](https://github.com/onuralpszr) Yes, I mean the augmentation function in python API. So if I want to enable data augmentation with default augmentation setting, do I need to set `augment=True` in train method explictly?
> 
> [ultralytics/ultralytics/cfg/default.yaml](https://github.com/ultralytics/ultralytics/blob/932c069b0273e97eea46745e3bc5bb73dcd9cecd/ultralytics/cfg/default.yaml#L65)
> 
> Line 65 in [932c069](/ultralytics/ultralytics/commit/932c069b0273e97eea46745e3bc5bb73dcd9cecd)
> 
>  augment: False # (bool) apply test-time augmentation during prediction

That's for "predict" not for train "https://docs.ultralytics.com/modes/predict/#inference-arguments"


augment | bool | False | Enables test-time augmentation (TTA) for predictions, potentially improving detection robustness at the cost of inference speed.
-- | -- | -- | --

**Comment by onuralpszr** [comment]
> [@onuralpszr](https://github.com/onuralpszr) Yes, I mean the augmentation function in python API. So if I want to enable data augmentation with default augmentation setting, do I need to set `augment=True` in train method explictly?
> 
> [ultralytics/ultralytics/cfg/default.yaml](https://github.com/ultralytics/ultralytics/blob/932c069b0273e97eea46745e3bc5bb73dcd9cecd/ultralytics/cfg/default.yaml#L65)
> 
> Line 65 in [932c069](/ultralytics/ultralytics/commit/932c069b0273e97eea46745e3bc5bb73dcd9cecd)
> 
>  augment: False # (bool) apply test-time augmentation during prediction

Also literally say "prediction" :)

**Comment by darouwan** [comment]
@onuralpszr  OK. Thanks. Just finding when I use it in train method, there is no error. so I'm worried about whether this parameter has any real impact on the training task.
